### PR TITLE
Add precision assertion to code intel regression test suites.

### PR DESCRIPTION
**Pull request by efritz**
Add precision flags to hovers, definitions, and reference assertions in code navigation tests.

This PR will allow us to collapse the overlapping concerns of codenav/codeintel regression tests as we now no longer need to test LSIF separately as we can now determine whether results are precise or fuzzy.

### CODE REVIEW

File: web/src/regression/util/codenav.ts [hunk 1]
```diff
@@ -84,21 +96,21 @@ async function getTooltip(driver: Driver): Promise<string> {
     return driver.page.evaluate(() => (document.querySelector('.e2e-tooltip-content') as HTMLElement).innerText)
 }
 
-async function collectLinks(driver: Driver): Promise<Set<string>> {
+async function collectLinks(driver: Driver): Promise<TestLocation[]> {
```
Comment (lguychard) [inline web/src/regression/util/codenav.ts]: Why the change from `Set` to `Array`?

Comment (efritz) [reply web/src/regression/util/codenav.ts]: There shouldn't be duplicates (although I'm not yet checking for that).

Comment (felixfbecker) [reply web/src/regression/util/codenav.ts]: If there are no duplicates, I would especially use a `Set` to communicate that it is a unique set (vs an ordered list with potential duplicates)

File: web/src/regression/util/codenav.ts [hunk 2]
```diff
@@ -115,10 +127,15 @@ async function getPanelTabTitles(driver: Driver): Promise<string[]> {
     ).map(normalizeWhitespace)
 }
 
-function collectVisibleLinks(driver: Driver): Promise<string[]> {
+function collectVisibleLinks(driver: Driver): Promise<TestLocation[]> {
     return driver.page.evaluate(() =>
         Array.from(document.querySelectorAll<HTMLElement>('.e2e-search-result')).flatMap(e =>
-            Array.from(e.querySelectorAll<HTMLElement>('a[href]')).map(a => a.getAttribute('href') || '')
+            Array.from(e.querySelectorAll<HTMLElement>('a[href]'))
+                .filter(a => a.nextElementSibling?.className.includes('file-match-children__item-badge-row'))
```
[the remark this answers is not in the document]
Comment (efritz) [reply web/src/regression/util/codenav.ts]: Updated.

File: web/src/regression/util/codenav.ts (hunk 1 above)
Comment (efritz) [reply web/src/regression/util/codenav.ts]: Reverting change.
